src/mount/readdata.cc

```diff
@@ -56,6 +57,60 @@
 #define EMPTY_REQUEST nullptr
 
 inline std::condition_variable readOperationsAvailable;
+inline int kMaxReadCacheRequestRetries = 10;
+
+std::unique_ptr<IMemoryInfo> createMemoryInfo() {
+    std::unique_ptr<IMemoryInfo> gMemoryInfo;
+    #ifdef _WIN32
+        gMemoryInfo = std::make_unique<WindowsMemoryInfo>();
+    #elif __linux__
+        gMemoryInfo = std::make_unique<LinuxMemoryInfo>();
+    #endif
+    return gMemoryInfo;
+}
+std::unique_ptr<IMemoryInfo> gMemoryInfo = createMemoryInfo();
+
+bool readShouldWaitForSystemMemory(size_t bytesToReadLeft) {
+	std::lock_guard lock(gUsedReadCacheMemoryMutex);
+	uint64_t avalaibleSystemMemory = gMemoryInfo->getAvailableMemory();
+	uint64_t virtualReadCacheAvailableMemory =
+		gReadCacheMaxSize.load() - gUsedReadCacheMemory;
+	return bytesToReadLeft > virtualReadCacheAvailableMemory ||
+			bytesToReadLeft > avalaibleSystemMemory;
+}
+
+void increaseUsedReadCacheMemory(size_t bytesToReadLeft) {
+	std::lock_guard lock(gUsedReadCacheMemoryMutex);
+	gUsedReadCacheMemory += bytesToReadLeft;
+}
+
+bool waitForEnoughSystemMemory(size_t bytesToReadLeft) {
+	bool readShouldWait = false;
+	if(readShouldWaitForSystemMemory(bytesToReadLeft)) {
+		safs_pretty_errlog(
+		    LOG_ERR,
+		    "Not enough system memory available for reading");
+		readShouldWait = true;
+	}
+	return readShouldWait;
+}
```

Comment:
Looks like this function is probably not needed anymore. Seems that it was used to log the error, but now you throw the exception in the function call.

Reply:
Done